python/plugins/processing/gui/wrappers.py

```diff
@@ -638,10 +641,15 @@ def _getOptions(self):
                 [QgsProcessingOutputRasterLayer,
                  QgsProcessingOutputMapLayer,
                  QgsProcessingOutputMultipleLayers])
+        elif self.parameterDefinition().layerType() == QgsProcessing.TypeMesh:
+            options = self.dialog.getAvailableValuesOfType(
+                (QgsProcessingParameterMeshLayer, QgsProcessingParameterMultipleLayers),
```

Comment:
Should also include QgsProcessingOutputMapLayer

Reply:
even there is no mesh write API?

Reply:
@nyalldawson should I add QgsProcessingOutputMapLayer even there is no write support?

Reply:
I guess not - we'll just have to remember to add this when mdal inevitably gets write support!